tmk_core/protocol/chibios/main.c

```diff
@@ -119,6 +119,8 @@ int main(void) {
   // TESTING
   // chThdCreateStatic(waThread1, sizeof(waThread1), NORMALPRIO, Thread1, NULL);
 
+  keyboard_setup();
```

Comment:
`matrix_setup` and `disable_jtag` already getting called. Doesn't this mean you've added a second call for these functions?

Reply:
Same comment for the other main.c

Reply:
Not sure what you mean here.

But `keyboard_pre_init_*` relies on `keyboard_setup.  Which doesn't exist.  And it should be the first thing called (or one of the first things).